core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -171,14 +171,23 @@ public void onTimelineChanged(Timeline timeline, @Nullable Object manifest, int
         if (advertHasNotStarted()) {
             adGroupIndex = player.getCurrentAdGroupIndex();
             adIndexInGroup = player.getCurrentAdIndexInAdGroup();
-            notifyAdvertStart(advertBreaks.get(adGroupIndex));
+
+            if (canPlayAdverts(adGroupIndex)) {
+                notifyAdvertStart(advertBreaks.get(adGroupIndex));
+            } else {
+                resetAdvertPosition();
```

Comment:
Need to do this otherwise when you get to the next advert, after skipping one, the position data will still be present. It will then try to mark the skipped advert as `Played` which leads to an `IllegalStateException`